packages/core/src/conversation/ConversationService/ConversationService.ts

```diff
@@ -388,50 +424,173 @@ export class ConversationService extends TypedEventEmitter<Events> {
+        if (!shouldRetry) {
+          this.logger.warn("Tried to remove users from MLS conversation but it's still broken after reset", error);
+          throw error;
+        }
+        this.logger.info(
+          "Tried to remove users from MLS conversation but it's broken, resetting the conversation",
+          error,
+        );
+        return this.handleBrokenMLSConversation(conversationId, newGroupId =>
+          this.removeUsersFromMLSConversation({
+            groupId: newGroupId,
+            conversationId,
+            qualifiedUserIds,
+            shouldRetry: false,
+          }),
+        );
+      }
 
-    //key material gets updated after removing a user from the group, so we can reset last key update time value in the store
-    await this.mlsService.resetKeyMaterialRenewal(groupId);
+      throw error;
+    }
+  }
 
-    return await this.getConversation(conversationId);
+  public async joinByExternalCommit(conversationId: QualifiedId, shouldRetry = true): Promise<void> {
+    try {
+      await this.mlsService.joinByExternalCommit(() => this.apiClient.api.conversation.getGroupInfo(conversationId));
+    } catch (error) {
+      if (MLSService.isBrokenMLSConversationError(error)) {
+        this.logger.info(
+          "Failed to join MLS conversation via external commit because it's broken, resetting the conversation",
+          error,
+        );
+        if (!shouldRetry) {
+          this.logger.warn("Tried to join MLS conversation but it's still broken after reset", error);
+          throw error;
+        }
+        return this.handleBrokenMLSConversation(conversationId, (_newGroupId: string) =>
+          this.joinByExternalCommit(conversationId, false),
+        );
+      }
+      throw error;
+    }
   }
 
-  public async joinByExternalCommit(conversationId: QualifiedId) {
-    return this.mlsService.joinByExternalCommit(() => this.apiClient.api.conversation.getGroupInfo(conversationId));
+  private async resetMLSConversation(conversationId: QualifiedId): Promise<BaseCreateConversationResponse> {
+    this.logger.info(`Resetting MLS conversation with id ${conversationId.id}`);
+
+    // STEP 1: Fetch the conversation to retrieve the group ID & epoch
+    const conversation = await this.apiClient.api.conversation.getConversation(conversationId);
+    const {group_id: groupId, epoch} = conversation;
+
+    if (!groupId || !epoch) {
+      const errorMessage = 'Could not find group id or epoch for the conversation';
+      this.logger.error(errorMessage, {conversationId});
+      throw new Error(errorMessage);
+    }
+
+    // STEP 2: Request backend to reset the conversation
+    this.logger.info(`Requesting backend to reset the conversation (group_id: ${groupId}, epoch: ${String(epoch)})`);
+    await this.apiClient.api.conversation.resetMLSConversation({
+      epoch,
+      groupId,
+    });
+
+    // STEP 3: Re-establish the conversation by re-adding all members
```

Comment:
Step 3 doesn't match what the code is doing. You are fetching the `selfUserQualifiedId`

Reply:
fixed in https://github.com/wireapp/wire-web-packages/pull/7405/commits/fd0656d770713c1a0dbb5ff66b50d2b1d53b3017